### PR TITLE
feat(state_store): refactor write stall to make code simple

### PR DESCRIPTION
**Pull request by Little-Wallace**
Signed-off-by: Little-Wallace <bupt2013211450@gmail.com>

I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

- Write stall (block write operation when buffer is full) does not need to be executed by flush thread. It could make code simple to read.
- Avoid maintain two similar code for memory control in both compactor-node and compute-node.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs [hunk 1]
```diff
@@ -93,11 +80,41 @@ impl SharedBufferBatch {
     pub fn new(
```
Comment (wenym1) [inline src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs:80]: We may rename it to `for_test` to avoid future misuse.

**Review by wenym1 — APPROVED** [review]
Rest LGTM!

**Review by Li0k — APPROVED** [review]
LGTM

File: src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs (hunk 1 above)
Comment (Little-Wallace) [reply src/storage/src/hummock/shared_buffer/shared_buffer_batch.rs]: fixed.

**Pull request merged**